core/src/main/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializer.scala

```diff
@@ -40,10 +41,16 @@ private[spark] class KVStoreProtobufSerializer extends KVStoreScalaSerializer {
 
 private[spark] object KVStoreProtobufSerializer {
 
- private[this] lazy val serializerMap: Map[Class[_], ProtobufSerDe] =
-   ServiceLoader.load(classOf[ProtobufSerDe])
-     .asScala.map(serDe => serDe.supportClass -> serDe).toMap
+  private[this] lazy val serializerMap: Map[Class[_], ProtobufSerDe[Any]] = {
+    def getGenericsType(klass: Class[_]): Class[_] = {
+      klass.getGenericInterfaces.head.asInstanceOf[ParameterizedType]
+        .getActualTypeArguments.head.asInstanceOf[Class[_]]
```

Comment:
```suggestion
        .getRawType.asInstanceOf[Class[_]]
```

Reply:
This is neat

Reply:
I want to double check, do you want to change it to

```
klass.getGenericInterfaces.head.asInstanceOf[ParameterizedType].getRawType.asInstanceOf[Class[_]]
```

Reply:
The result of 
```
klass.getGenericInterfaces.head.asInstanceOf[ParameterizedType]
  .getRawType.asInstanceOf[Class[_]]
```
is always `org.apache.spark.status.protobuf.ProtobufSerDe`

Reply:
The result of 
```
klass.getGenericInterfaces.head.asInstanceOf[ParameterizedType]
  .getActualTypeArguments.head.asInstanceOf[Class[_]]
```
are 

```
org.apache.spark.status.RDDStorageInfoWrapper
org.apache.spark.status.ApplicationInfoWrapper
org.apache.spark.status.ApplicationEnvironmentInfoWrapper
org.apache.spark.status.CachedQuantile
org.apache.spark.status.ExecutorStageSummaryWrapper
org.apache.spark.status.StreamBlockData
org.apache.spark.status.TaskDataWrapper
org.apache.spark.status.JobDataWrapper
org.apache.spark.status.ResourceProfileWrapper
org.apache.spark.status.SpeculationStageSummaryWrapper
org.apache.spark.status.ExecutorSummaryWrapper
org.apache.spark.status.ProcessSummaryWrapper
org.apache.spark.status.RDDOperationGraphWrapper
org.apache.spark.status.StageDataWrapper
```
This is what we need

Reply:
You are right.